lib/addons/velero/index.ts

```diff
@@ -0,0 +1,239 @@
+import { ClusterAddOn, ClusterInfo } from "../../spi";
+import { Constants } from "..";
+import * as s3 from "@aws-cdk/aws-s3";
+import * as iam from "@aws-cdk/aws-iam";
+import { GatewayVpcEndpointAwsService } from "@aws-cdk/aws-ec2";
+import deepmerge = require('deepmerge');
```

Comment:
@askulkarni2  the spread operator wont work as it will only work as the 1st layer of json documents. 
We have to use deepmerge, otherwise it wont be able to deeply merge.

Reply:
example:
```json
    values:{
        initContainers:[
            {
                name: "velero-plugin-for-aws",
                image: "velero/velero-plugin-for-aws:v1.2.0",
                imagePullPolicy: "IfNotPresent",
                volumeMounts:[
                    {
                        mountPath: "/target",
                        name: "plugins"
                    }
                ]
            }
        ],
        configuration: {
            provider: "aws",
            backupStorageLocation:{
                name: "default",
                config:{}
            },
            volumeSnapshotLocation:{
                name: "default",
                config:{}
            },
        },
        serviceAccount: {
            server:{}
        }
```

Reply:
Thanks @shapirov103  did not know the `ts-deepmerge` before. Thanks for letting me know and update the package accordingly.